core-java-modules/core-java-concurrency-advanced-4/src/main/java/com/baeldung/producerconsumer/DataQueue.java

```diff
@@ -14,22 +14,26 @@ public class DataQueue {
     }
 
     public boolean isFull() {
-        return queue.size() == maxSize;
+        synchronized (queue) {
+            return queue.size() == maxSize;
+        }
     }
 
     public boolean isEmpty() {
-        return queue.isEmpty();
+        synchronized (queue) {
+            return queue.isEmpty();
+        }
     }
 
     public void waitOnFull() throws InterruptedException {
         synchronized (FULL_QUEUE) {
-            FULL_QUEUE.wait();
+            FULL_QUEUE.wait(1000);
```

Comment:
Rollback any changes used for testing requester's claim

Reply:
Done